src/documentation/sdks/euphoria.md

```diff
@@ -521,6 +517,21 @@ Dataset<String> animals =
 // animal will contain: "cheetah", "cat", "lynx", "jaguar", "squirrel", "mouse", "rat", "lemming", "beaver"
 ```
 
+### `TopPerKey`
+Emits one top-rated element per key. Key of type `K` is extracted by given `UnaryFunction`. Another `UnaryFunction` extractor allows for conversion input elements to values of type `V`. Selection of top element is based on _score_, which is obtained from each element by user supplied `UnaryFunction` called score calculator. Score type is denoted as `ScoreT` and it is required to extend `Comparable<ScoreT>` so scores of two elements can be compared directly. Output dataset elements are of type `Triple<K, V, ScoreT>`.
+```java
+// suppose 'animals contain: [ "mouse", "elk", "rat", "mule", "elephant", "dinosaur", "cat", "duck", "caterpillar" ]
+Dataset<Triple<Character, String, Integer>> longestNamesByLetter =
+  TopPerKey.named("longest-animal-names")
+    .of(animals)
+    .keyBy(name -> name.charAt(0)) // first character is the key
+    .valueBy(UnaryFunction.identity()) // value type is the same as input element type
+    .scoreBy(String::length) // length defines score, note that Integer implements Comparable<Integer>
+    .output();
+//longestNamesByLetter wil contain: [ ('m', "mouse", 5), ('r', "rat", 3), ('e', "elephant", 8), ('d', "dinosaur", 8), ('c', "caterpillar", 11) ]
+```
+`TopPerKey` is a shuffle operator so it allows for widowing to be defined.
```

Comment:
I'd suggest: `TopPerKey` is a stateful operator, so it needs windowing to be defined. Shuffle doesn't imply windowing, it is actually the state that does.

Reply:
The actually correct definition is "either windowing or non-default trigger has to be defined", but that applies on all places in Euphoria (not sure if we actually support non-default triggers on global window, yet).